base/src/main/java/io/spine/query/ComparisonOperator.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2020, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.query;
+
+import com.google.protobuf.Timestamp;
+import com.google.protobuf.util.Timestamps;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.util.Objects;
+
+import static java.lang.String.format;
+
+/**
+ * Defines how the queried records are compared against the desired parameter values.
+ *
+ * <h1><a name="supported_types"><strong>Supported field types</strong></a></h1>
+ *
+ * <p>The equality comparisons support the fields of all types. The operation is performed
+ * via the {@link Objects#equals} method. A {@code null} reference is considered equal
+ * to another {@code null} reference.
+ *
+ * <p>Order-based comparison supports only the values of {@code Comparable} types and
+ * {@link com.google.protobuf.Timestamp}s. When trying to compare unsupported types,
+ * an {@code UnsupportedOperationException} is thrown.
+ *
+ * <p>It is required that the runtime Java class of the two compared values is the same. Otherwise,
+ * an {@code IllegalArgumentException} is thrown.
+ */
+public enum ComparisonOperator {
+
+    /**
+     * The actual value must be equal to the value of the subject parameter.
+     */
+    EQUALS {
+        @Override
+        public boolean eval(@Nullable Object left, @Nullable Object right) {
+            return Objects.equals(left, right);
+        }
+    },
+
+    /**
+     * The actual value must be less than the value of the subject parameter.
+     */
+    LESS_THAN {
+        @Override
+        public boolean eval(@Nullable Object left, @Nullable Object right) {
+            return GREATER_THAN.eval(right, left);
```

Comment:
```suggestion
            return !EQUALS(right, left) && !GREATER_THAN.eval(right, left);
```

Reply:
The operands were swapped places, therefore the original code works properly.